mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java

```diff
@@ -165,7 +165,7 @@ void storeAndUpdateWorkers(final IMantisWorkerMetadata existingWorker, final IMa
 
     List<JobArtifact> listJobArtifacts(String name, String version) throws IOException;
 
-    List<String> listJobArtifactsByName(String prefix) throws IOException;
+    List<String> listJobArtifactsByName(String prefix, String contains) throws IOException;
```

Comment:
I wonder if we should either have two different APIs or one API with one parameter that works for both cases. It feels like we want some fuzzy search which, in my understanding, would require just one parameter.

Reply:
Good point. I was thinking to add a new param instead of renaming it because:
- keep backward comp with existing endpoint
- prefix and contains can have different implementations and prefix-based search is more efficient than `contains`.